apis/field_error.go

```diff
@@ -285,7 +375,11 @@ func merge(errs []*FieldError) []*FieldError {
 
 // key returns the key using the fields .Message and .Details.
 func key(err *FieldError) string {
-	return fmt.Sprintf("%s-%s", err.Message, err.Details)
+	l := err.Level
+	if l == "" {
+		l = ErrorLevel
+	}
```

Comment:
If you switch to Stringer, I think you could remove these lines.

Reply:
At first I thought you meant `fmt.Stringer`, but I see you mean a tool (which I assume auto-generates the method, TIL).  I think for this to work, `ErrorLevel` would have to be explicitly `0` and then subsequent values would start with `iota`, or something like that... 🤔

Reply:
[`iota` will start at 0](https://github.com/golang/go/wiki/Iota), so this would mean `ErrorLevel` was 0 and `WarningLevel` was 1.

Since we don't serialize these values, if we wanted to insert an "AlarmLevel" between "Error" and "Warning", we could redefine "Warning" to be 2 and "Alarm" to be 1. (I can't imagine that we'll need multiple levels of warnings, but I've implemented a lot of code I never imagined I'd need to write...)

Reply:
Interesting, I always thought of it like an "epsilon" value, but perhaps it was designed to start at zero for cases exactly like this.  I'll try to experiment with this in a bit 🤞

Reply:
https://github.com/knative/pkg/pull/2511